app/models/mdm/service.rb

```diff
@@ -222,6 +222,15 @@ class Mdm::Service < ActiveRecord::Base
                 in: PROTOS
             }
 
+  validates :host_id,
```

Comment:
@dmaloney-r7 We need to add a unique index on service on host_id, port, proto, otherwise we could run into multiple threads trying to insert duplicates and being able to succeed: http://robots.thoughtbot.com/the-perils-of-uniqueness-validations